mlflow/torchscript.py

```diff
@@ -0,0 +1,108 @@
+from pathlib import Path
+import logging
+import yaml
+
+import torch
+import pandas as pd
+import numpy as np
+import mlflow
+from mlflow import pyfunc
+from mlflow.models import Model
+from mlflow.exceptions import MlflowException
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils.model_utils import _get_flavor_configuration
+
+FLAVOR_NAME = "torchscript"
+_SERIALIZED_TORCH_MODEL_FILE_NAME = "model.pt"
+_logger = logging.getLogger(__name__)
+
+
+def get_default_conda_env():
+    """
+    :return: The default Conda environment for MLflow Models produced by calls to
+             :func:`save_model()` and :func:`log_model()`.
+    """
+    import torch
+
+    return _mlflow_conda_env(
+        additional_conda_deps=[
+            "pytorch={}".format(torch.__version__),
```

Comment:
Is it sufficient to install just `pytorch`? From the [pytorch homepage](https://pytorch.org/) it seems for conda installations we need to also install `torchvision`, and if on linux or windows to install `cpuonly` as well

Reply:
e.g. in the [pytorch flavor](https://github.com/mlflow/mlflow/blob/7969f9140874e0644770f255d106f7a729150aad/mlflow/pytorch/__init__.py#L49-L51), we also install `torchvision`

Reply:
`torchvision` is required only for `vision` tasks, so it's not a mandatory requirement for pytorch
`cpuonly` is required to choose the non-CUDA version of pytorch. Perhaps it makes more sense to keep the CPU only version default. I'll add that

Reply:
Sounds good, yeah let me know if you hit issues related to this! How expensive of a dependency to install is `torchvision`? Given that we install it in the conda env for the `pytorch` flavor, it would be nice to install it here as well for consistency (otherwise users may be confused that they need to specify the extra `torchvision` dependency when using `mlflow.torchvision.log_model` but not when using `mlflow.pytorch.log_model`)

Reply:
Fair enough